src/test/kotlin/org/rust/ide/surroundWith/RustSurrounderTestCaseBase.kt

```diff
@@ -0,0 +1,53 @@
+package org.rust.ide.surroundWith
+
+import com.intellij.codeInsight.generation.surroundWith.SurroundWithHandler
+import com.intellij.lang.LanguageSurrounders
+import com.intellij.lang.surroundWith.Surrounder
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.util.readText
+import org.assertj.core.api.Assertions.assertThat
+import org.rust.lang.RustLanguage
+import org.rust.lang.RustTestCaseBase
+import java.nio.file.Paths
+import java.util.*
+
+abstract class RustSurrounderTestCaseBase(val surrounder: Surrounder) : RustTestCaseBase() {
+    override val dataPath: String get() = "org/rust/ide/surroundWith/fixtures/$testClassName"
+
+    protected val testClassName: String =
+        surrounder.javaClass.simpleName.replace(Regex("^RustWith(\\w+)Surrounder$"), "$1").decapitalize()
+
+    protected fun doTest(isApplicable: Boolean = true) {
```

Comment:
Perhaps create two different functions instead of one with a boolean parameter, which is always a compile time constant?